### PR TITLE
feat: clone the registry archive and minor fixes

### PR DESCRIPTION
**Pull request by henilp105**
- [x] feat: add enter key for search functionality
- [x] feat: sort packages by the order of their versions
- [x] feat: add registry archive functionality
- [x] feat: user can login by username or email and password
- [x] feat: added docs for the env variables
- [x] fix: minor bug fixes

Thanks and Regards,
Henil  

CC: @awvwgk @arteevraina @minhqdao @perazz

### CODE REVIEW

File: flask/auth.py [hunk 1]
```diff
@@ -43,11 +43,14 @@ def generate_uuid():
 @swag_from("documentation/login.yaml", methods=["POST"])
 def login():
     salt = env_var["salt"]
-    email = request.form.get("email")
+    username = request.form.get("email")
```
Comment (arteevraina) [inline flask/auth.py:46]: The form-data key that is "email" here is indicating that we only require email and not username in the code.

So, can you name it something more generic on both flask and react side.

For example this.

```
user_identifier = require.form.get("user_identifier")
```

And then,

```
query = {"$and": [{"$or": [{"email": user_identifier}, {"username":user_identifier}]}, {"password": hashed_password}]}
```

File: flask/mongo.py [hunk 2]
```diff
@@ -1,16 +1,55 @@
...
+    mongo_password = os.environ['MONGO_PASSWORD']
     client = MongoClient(mongo_uri)
 except KeyError as err:
     print("Add MONGO_URI to .env file")
 
 db = client[database_name]
 file_storage = GridFS(db, collection="tarballs")
+
+
+@app.route("/registry/clone", methods=["GET"])
```
Comment (arteevraina) [inline flask/mongo.py:24]: Can you test if this API crashes in vercel deployment ?
Also, should be consider making this API authorized ?

**Review by arteevraina — CHANGES_REQUESTED** [review]
Thank you @henilp105 . PR looks good to me. I have few suggestions & questions commented.

File: flask/mongo.py (hunk 2 above)
Comment (henilp105) [reply flask/mongo.py:24]: yes, as vercel has headless running thus as we are storing the archive in a static folder , I have tested It runs well in the docker containers. I think according to the general requirements of the https://repology.org/docs/requirements , we should make this open to everyone.

File: flask/auth.py (hunk 1 above)
Comment (henilp105) [reply flask/auth.py:46]: sure.

**Review by arteevraina — APPROVED** [review]

**Comment by arteevraina** [comment]
LGTM @henilp105

**Comment by henilp105** [comment]
Sure, will patch it.

**Pull request merged**